### DISCUSSION TITLE
error C2059: syntax error: ')'

### DISCUSSION THREAD

**Discussion by Megidd** [AcademySoftwareFoundation/openvdb#1302] [Q&A] [answered]
I built OpenVDB by these instructions:

https://github.com/AcademySoftwareFoundation/openvdb#windows

Then I'm using OpenVDB in my code. My `CMakeLists.txt` contains:

```cmake
include_directories("C:/Program Files (x86)/OpenVDB/include")
```

But, when building my code, I'm receiving such errors:

```
C:\Program Files (x86)\OpenVDB\include\openvdb/math/BBox.h(62,18): error C2059: syntax error: ')' [C:\Users\m3\repos\cpp-service\build\cpp-service.vcxproj]
C:\Program Files (x86)\OpenVDB\include\openvdb/math/BBox.h(140): message : see reference to class template instantiation 'openvdb::v7_2::math::BBox<Vec3T>'
being compiled [C:\Users\m3\repos\cpp-service\build\cpp-service.vcxproj]
```

### Versions
I have used different OpenVDB tags, all of them throw the same error:

```bash
git checkout v8.2.0
git checkout v8.1.0
git checkout v7.2.3
```

### Toolchains
I have used both Visual Studio 2017 and 2019 toolchains. But they both throw the same errors.

### Cause
It looks like a compiler related error. But I ran out of options to solve it. Could anyone help?

Some months ago, I was building the same code just fine 😕

**Comment by Idclip** [comment, marked as answer]
Windows is most likely redefining min/max functions as macros which causes this. OpenVDB tells windows not to define these in two places; the first is in `Platform.h` which `BBox.h` includes. The second is in the distributed CMake modules which it doesn't seem like you're using. There are two solutions:

1) Use the installed CMake modules. see: https://www.openvdb.org/documentation/doxygen/build.html#buildUsingOpenVDB
2) Make sure you include VDB headers first OR `#define NOMINMAX` before you include windows system headers. For example:

works:
```
#include <openvdb.h> // defines platform requirements needed for VDB to build
#include <windows.h> 
```

doesn't work:
```
#include <windows.h> // could also be indirectly included through another header
#include <openvdb.h> // oops, msvc has already brought in the badness
```
